src/componets/SearchNav/index.jsx

```diff
@@ -0,0 +1,151 @@
+      { name: 'Speed-Boost', active: false },
+      { name: 'Battle-armor', active: false },
+      { name: 'Sturdy', active: false },
+      { name: 'Damp', active: false },
+      { name: 'Limber', active: false },
+      { name: 'Sand-veil', active: false },
+    ],
+    weight: [
+      { name: 'Light', active: false },
+      { name: 'Medium', active: false },
+      { name: 'Heavy', active: false },
+    ],
+    type: [
+      { name: 'Bug', active: false },
+      { name: 'Dark', active: false },
+      { name: 'Dragon', active: false },
+      { name: 'Electric', active: false },
+      { name: 'Fairy', active: false },
+      { name: 'Fighting', active: false },
+      { name: 'Fire', active: false },
+      { name: 'Flying', active: false },
+      { name: 'Ghost', active: false },
+      { name: 'Normal', active: false },
+      { name: 'Water', active: false },
+      { name: 'Grass', active: false },
+      { name: 'Ground', active: false },
+      { name: 'Ice', active: false },
+      { name: 'Poison', active: false },
+      { name: 'Psychic', active: false },
+      { name: 'Rock', active: false },
+      { name: 'Steel', active: false },
+    ],
+    height: [
+      { name: 'Small', active: false },
+      { name: 'Medium', active: false },
+      { name: 'Large', active: false },
+    ],
+  };
+  const [order, setOrder] = useState([
+    { name: 'Ascending', active: true },
+    { name: 'Descending', active: false },
+    { name: 'A-Z', active: false },
+    { name: 'Z-A', active: false },
+  ]);
+  const [isActiveBtn, setActiveBtn] = useState(false);
+  const [data, setData] = useState({ ...localData });
+  const clean = () => {
+    setData({ ...localData });
+    setOrder([
+      { name: 'Ascending', active: true },
+      { name: 'Descending', active: false },
+      { name: 'A-Z', active: false },
+      { name: 'Z-A', active: false },
+    ]);
+  };
+  useEffect(() => {
+    const allObjects = Object.keys(data).map(
+      (item) => data[item].some((element) => element.active),
+    );
+    setActiveBtn(allObjects.some((item) => item));
+  }, [data]);
+  const sendData = () => {
+    const filterData = Object.keys(data).map(
+      (item) => data[item].filter((element) => element.active),
+    );
+    // eslint-disable-next-line no-console
+    console.log(filterData);
+  };
+  const updateData = (property, index) => {
+    const updatingData = ({ ...data });
+    if (property === 'ability') {
+      updatingData.ability = data.ability.map((item) => ({ name: item.name, active: false }));
+    }
+    updatingData[property][index].active = !data[property][index].active;
+    setData({ ...updatingData });
+  };
+  const updateOrder = (index) => {
+    const updatingData = order.map((item) => ({ name: item.name, active: false }));
+    updatingData[index].active = !order[index].active;
+    const somethingTrue = updatingData.some((item) => item.active);
+    if (!somethingTrue) updatingData[index].active = true;
```

Comment:
not sure what is this for

Reply:
I added that part because when you selected the same item in the order dropdown it would remain blank so what it does it leave the same selected item and not remain blank